apis/database/v1beta1/sql_types.go

```diff
@@ -179,6 +184,18 @@ type SQLServerParameters struct {
 	StorageProfile StorageProfile `json:"storageProfile"`
 }
 
+// CreateMode controls the creation behaviour
+// +kubebuilder:validation:Enum=Default;GeoRestore;PointInTimeRestore;Replica;ServerPropertiesForCreate
```

Comment:
Can you use kubebuilder validation comments on a type declaration like this? I've only ever seen them used on fields.

Reply:
Yep; see https://book.kubebuilder.io/cronjob-tutorial/api-design.html for instance. And of course the generated code